blog/02-18-2023-Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain/Build-an-NFT-Full-Stack-Monster-War-Game-on-The-Celo-Blockchain.md

```diff
@@ -0,0 +1,998 @@
+    return owner;
+  } catch (e) {
+    console.log({ e });
+  }
+};
+ ```
+
+The `fetchNftContractOwner()` function is used to get the address that deployed the NFT contract. It takes the `minterContract` as an argument and returns the address.
+
+```javascript
+export const swallow = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        console.log(minterContract, index);
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .swallowNFT(index)
+          .send({ from: defaultAccount });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+```
+
+The `swallow()` function is used to swallow an NFT. It takes the `minterContract` and `performActions` as arguments, along with the index of the NFT. It then uses the `defaultAccount` from the kit to call the `swallowNFT` method and swallow the NFT.
+
+```javascript
+export const upgrade = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        const price = ethers.utils.parseUnits(String(0.5), "ether");
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .upgradeNFT(index)
+          .send({ from: defaultAccount, value: price });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+```
+
+The `upgrade()` function is used to upgrade an NFT. It takes the `minterContract` and `performActions` as arguments, along with the index of the NFT. It then uses the `defaultAccount` from the kit to call the `upgradeNFT` method and upgrade the NFT. 
+
+```javascript
+export const remove = async (minterContract, performActions, index) => {
+  try {
+    await performActions(async (kit) => {
+      try {
+        const { defaultAccount } = kit;
+        await minterContract.methods
+          .remove(index)
+          .send({ from: defaultAccount });
+      } catch (error) {
+        console.log({ error });
+      }
+    });
+  } catch (error) {
+    console.log(error);
+  }
+};
+```
+
+The `remove()` function is used to remove an NFT. It takes the `minterContract` and `performActions` as arguments, along with the index of the NFT. It then uses the `defaultAccount` from the kit to call the remove method and remove the NFT.
+
+There you have it. We have successfully interacted with our smart contract.
+
+## Next Steps
+
+I hope you learned a lot from this tutorial. Here are some relevant links that would aid your learning further.
+
+- <https://docs.celo.org/>
+- <https://solidity-by-example.org/>
+- <https://www.openzeppelin.com/contracts>
+- <https://docs.soliditylang.org/en/v0.8.17/>
+
+## About the author
+
+I'm Jonathan Iheme, A full stack block-chain Developer from Nigeria.
+
+## References
+
+- <https://web3.storage/docs/>
```

Comment:
can you embed these link in a text for example "click [here](https://github.com/celo-org/docs/pull/694/www.link.com)"

Reply:
it's important for people to see the name of the reference links

Reply:
it's important for people to see the name of the reference links